### PR TITLE
submitted an initial solution to week 7

### PR DESCRIPTION
**Pull request by Andrey8860**

### CODE REVIEW

File: Week 7/solution/src/test/java/com/epam/week7/pages/SportchekHeader.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class SportchekHeader extends BasePage {
+	
+	// In this case we use locators and getter methods, because Pencil Banner itself may be used not only in the order
+	// of sign in and then sign out. Due to this, we can never know before runtime if we should initialize 
+	// the necessary elements
+	private final String signInLinkLocator = ".nl-sm-block .nl-pencil-banner__login";
+	private final String signOutButtonLocator = "//div[@class='nl-pencil-banner__sign-out']/button";
+	private final String hiUsernameLinkLocator = ".nl-sm-block #accountButton";
+	
+	@FindBy(xpath="(//span[@class='nl-store-locator--selected-store'])[2]")
```
Comment (mamax) [inline Week 7/solution/src/test/java/com/epam/week7/pages/SportchekHeader.java:19]: Could you change this locator to something else without using `[2]`, pls?

File: Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		SportchekLoginModal sportchekLoginModal = new SportchekLoginModal(driver);
+		sportchekLoginModal.enterLoginEmail(loginEmail);
+		sportchekLoginModal.enterLoginPassword(loginPassword);
+		sportchekLoginModal.clickSignInButton();
+		sportchekLoginModal.waitForThePageSpinnerToBeInvisible();
+		
+		sportchekHeader.clickHiUsernameLink();
+		sportchekHeader.clickSignOutButton();
+		
+		Assert.assertTrue(sportchekHeader.getSignInLink().isDisplayed(), "Sign out was unsuccessful");
```
Comment (mamax) [inline Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java:37]: you may extract constant like
private final String MESSAGE = "";

File: Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		storeLocatorPage.searchForStoreWithFirstOptionSelected(storeSearchQuery);
+		storeLocatorPage.setPreferredStoreAtIndex(0);
+		storeLocatorPage.clickOnStoreDetailsLinkAtIndex(0);
+		
+		SportchekStoreDetailsPage storeDetailsPage = new SportchekStoreDetailsPage(driver);
+		SportchekHeader sportchekHeader = new SportchekHeader(driver);
+		
+		Assert.assertTrue(storeDetailsPage.getStoreTitleInDetailedAddressSection().getText()
+				.equalsIgnoreCase(sportchekHeader.getPreferredStoreInHeader().getText()),
+				"Change of preferred store through store locator was unsuccessful");
```
Comment (mamax) [inline Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java:61]: use some constant

File: Week 7/solution/src/test/java/com/epam/week7/pages/SportchekStoreLocatorPage.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		PageFactory.initElements(driver, this);
+	}
+	
+	public void searchForStoreWithFirstOptionSelected(String searchQuery) {
+		enterStoreSearchQuery(searchQuery);
+		clickStoreSearchButton();
+		selectAutocompleteSearchResultAtIndex(0);
+		clickStoreSearchButton();
+		
+		//Thread.sleep() is not ok here, but I want to discuss how we can deal with refresh of elements, since I was
```
Comment (mamax) [inline Week 7/solution/src/test/java/com/epam/week7/pages/SportchekStoreLocatorPage.java:41]: create separate class  and try s-ng like :
```
public static void waitForPageReadyState() {
        new WebDriverWait(getWebDriver(), Duration.ofSeconds(30)).until(driver ->
                "complete".equals(Selenide.executeJavaScript("return document.readyState")));
    }
```

File: Week 7/solution/src/test/java/com/epam/week7/pages/SportchekHeader.java (hunk 1 above)
Comment (Andrey8860) [reply Week 7/solution/src/test/java/com/epam/week7/pages/SportchekHeader.java]: Resolved

File: Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java (hunk 2 above)
Comment (Andrey8860) [reply Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java]: Resolved

File: Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java (hunk 3 above)
Comment (Andrey8860) [reply Week 7/solution/src/test/java/com/epam/week7/tests/SportchekTests.java]: Resolved

File: Week 7/solution/src/test/java/com/epam/week7/pages/SportchekStoreLocatorPage.java (hunk 4 above)
Comment (Andrey8860) [reply Week 7/solution/src/test/java/com/epam/week7/pages/SportchekStoreLocatorPage.java]: Resolved. Thank you for the suggestion.
Please note that I did NOT create a separate class for Waits. This was actually done in the later modules, as part of refactoring. I noticed the same problem by myself when learning about SOLID principles and I modified it, however in module 10

**Pull request merged**